src/node_osrm_support.hpp

```diff
@@ -420,13 +420,42 @@ inline bool parseCommonParameters(const v8::Local<v8::Object> &obj, ParamType &p
     if (obj->Has(Nan::New("annotations").ToLocalChecked()))
     {
         auto annotations = obj->Get(Nan::New("annotations").ToLocalChecked());
-        if (annotations->IsBoolean())
+
+        std::string annotations_str = *v8::String::Utf8Value(annotations);
+        if (annotations_str == "true")
+        {
+            params->annotations_type = osrm::RouteParameters::AnnotationsType::All;
+            params->annotations = true;
+        }
+        else if (annotations_str == "false")
+        {
+            params->annotations_type = osrm::RouteParameters::AnnotationsType::None;
+            params->annotations = false;
+        }
+        else if (annotations_str == "duration")
```

Comment:
The Javascript way here would be to have a check if the input is an array or a boolean.

Like `osrm.route({coordinates: ...., annotations: ["duration", "distance"]})` and  `osrm.route({coordinates: ...., annotations: false})`